jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererAutoTriggerService.kt

```diff
@@ -212,28 +238,81 @@ class CodeWhispererAutoTriggerService : CodeWhispererAutoTriggerHandler, Disposa
+                if (previousOneDecision == CodewhispererPreviousSuggestionState.Reject) {
+                    if (isExperimentGroup) {
+                        CodeWhispererClassifierConstants.prevDecisionRejectCoefficientExp
+                    } else CodeWhispererClassifierConstants.prevDecisionRejectCoefficient
+                } else {
+                    0.0
+                }
             previousOneOther =
                 if (
                     previousOneDecision != CodewhispererPreviousSuggestionState.Accept &&
                     previousOneDecision != CodewhispererPreviousSuggestionState.Reject
                 ) {
-                    CodeWhispererClassifierConstants.prevDecisionOtherCoefficient
+                    if (isExperimentGroup) {
+                        CodeWhispererClassifierConstants.prevDecisionOtherCoefficientExp
+                    } else CodeWhispererClassifierConstants.prevDecisionOtherCoefficient
                 } else {
                     0.0
                 }
         }
 
+        var leftContextLengthCoefficient: Double = 0.0
+        if (isExperimentGroup) {
+            leftContextLengthCoefficient = when (leftContextLength) {
+                in 0..4 -> CodeWhispererClassifierConstants.lengthLeft0To5Exp
+                in 5..9 -> CodeWhispererClassifierConstants.lengthLeft5To10Exp
+                in 10..19 -> CodeWhispererClassifierConstants.lengthLeft10To20Exp
+                in 20..29 -> CodeWhispererClassifierConstants.lengthLeft20To30Exp
+                in 30..39 -> CodeWhispererClassifierConstants.lengthLeft30To40Exp
+                in 40..49 -> CodeWhispererClassifierConstants.lengthLeft40To50Exp
+                else -> 0.0
+            }
+        }
+
+        val normalizedLengthOfRight = if (isExperimentGroup) {
+            CodeWhispererClassifierConstants.lengthofRightCoefficientExp * VariableTypeNeedNormalize.LenRight.normalizeExp(lengthOfRight.toDouble())
+        } else CodeWhispererClassifierConstants.lengthofRightCoefficient * VariableTypeNeedNormalize.LenRight.normalize(lengthOfRight.toDouble())
+
+        val normalizedLengthOfLeftCurrent = if (isExperimentGroup) {
+            CodeWhispererClassifierConstants.lengthOfLeftCurrentCoefficientExp *
+                VariableTypeNeedNormalize.LenLeftCur.normalizeExp(lengthOfLeftCurrent.toDouble())
+        } else CodeWhispererClassifierConstants.lengthOfLeftCurrentCoefficient * VariableTypeNeedNormalize.LenLeftCur.normalize(lengthOfLeftCurrent.toDouble())
+
+        val normalizedLengthOfPrev = if (isExperimentGroup) {
+            CodeWhispererClassifierConstants.lengthOfLeftPrevCoefficientExp * VariableTypeNeedNormalize.LenLeftPrev.normalizeExp(lengthOfLeftPrev)
+        } else CodeWhispererClassifierConstants.lengthOfLeftPrevCoefficient * VariableTypeNeedNormalize.LenLeftPrev.normalize(lengthOfLeftPrev)
```

Comment:
all these look hard to maintain. Can you abstract this and get instances that would return different values based on the `isExperimentGroup` value?

Reply:
yeah it's not very pretty will see if I can refactor this, but this is also temporary

Reply:
meanwhile I think it's not a blocker.